src/__tests__/custom-document/custom-document.test.tsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import '@testing-library/jest-dom';
+import { render, screen } from '@testing-library/react';
+import { getPage } from '../../index';
+import CustomDocumentWithGIP_Page from './__fixtures__/custom-document-with-gip/pages/page';
+import CustomApp from './__fixtures__/custom-document-with-gip/pages/_app';
+
+describe('_document support', () => {
+  describe('_document with getInitialProps', () => {
+    describe('with custom _app', () => {
+      it('renders page wrapped in custom _app wrapped in _document', async () => {
+        const { page } = await getPage({
+          nextRoot: __dirname + '/__fixtures__/custom-document-with-gip',
+          route: '/page',
+          useDocument: true,
+        });
+        const { container, debug } = render(page);
+        const head = container.querySelector('head') as HTMLHeadElement;
+        const html = container.querySelector('html') as HTMLHtmlElement;
+        expect(html).toHaveAttribute('lang', 'en');
+        expect(head.querySelector('meta[name="Description"]')).toHaveAttribute(
+          'Content',
+          'Custom document description'
+        );
+        const actual = container.querySelector('#__next') as HTMLDivElement;
+        actual.removeAttribute('id');
+
+        const { container: expected } = render(
+          <CustomApp Component={CustomDocumentWithGIP_Page} pageProps={{}} />,
+          { container: document.createElement('html') }
+        );
+
+        // @NOTE this match produces a circular structure error
+        // https://github.com/facebook/jest/issues/10577
+        // expect(actual).toEqual(expected);
+        expect(true).toBe(true);
```

Comment:
Does this happen if you remove the custom 'html' container as well? 🤔

Reply:
No! It seems to only happen with the element extracted from `#__next` element. With `container` it works fine.

I also noticed that the element has an unexpected comment created by Next.js:
```js
<div>
  _app
  <!-- -->
  /custom-document-with-gip/page
</div>
```